contracts/BoostedMasterChefJoe.sol

```diff
@@ -0,0 +1,589 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/utils/EnumerableSet.sol";
+import "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import "./interfaces/IERC20.sol";
+
+interface IMasterChef {
+    struct UserInfo {
+        uint256 amount; // How many LP tokens the user has provided.
+        uint256 rewardDebt; // Reward debt. See explanation below.
+    }
+
+    struct PoolInfo {
+        IERC20 lpToken; // Address of LP token contract.
+        uint256 allocPoint; // How many allocation points assigned to this pool. JOE to distribute per block.
+        uint256 lastRewardTimestamp; // Last block number that JOE distribution occurs.
+        uint256 accJoePerShare; // Accumulated JOE per share, times 1e12. See below.
+    }
+
+    function poolInfo(uint256 pid) external view returns (IMasterChef.PoolInfo memory);
+
+    function totalAllocPoint() external view returns (uint256);
+
+    function joePerSec() external view returns (uint256);
+
+    function deposit(uint256 _pid, uint256 _amount) external;
+
+    function devPercent() external view returns (uint256);
+
+    function treasuryPercent() external view returns (uint256);
+
+    function investorPercent() external view returns (uint256);
+}
+
+interface IRewarder {
+    function onJoeReward(address user, uint256 newLpAmount) external;
+
+    function pendingTokens(address user) external view returns (uint256 pending);
+
+    function rewardToken() external view returns (IERC20);
+}
+
+library BoringERC20 {
```

Comment:
Yeah this is a great call, I have pushed these changes. Annoyingly there are already interfaces there but because of dependency issues I had to basically duplicate the files and change the names. It's not a great solution but at least this file is cleaned up now.